microsoft/testsuites/nvme/nvme.py

```diff
@@ -77,3 +92,252 @@ def nvme_basic_validation(self, environment: Environment, node: Node) -> None:
+
+            # 2. Umount the mountpoint.
+            mount.umount(disk_name=namespace, point=mount_point, erase=False)
+
+            # 3. Run blkdiscard command on the partition.
+            blkdiscard = node.execute(
+                f"blkdiscard -v {namespace}p1", shell=True, sudo=True
+            )
+            if 0 != blkdiscard.exit_code:
+                blkdiscard = node.execute(
+                    f"blkdiscard -f -v {namespace}p1", shell=True, sudo=True
+                )
+            blkdiscard.assert_exit_code()
+
+            # 4. Remount command should fail after run blkdiscard command.
+            mount_result = node.execute(
+                f"mount {namespace}p1 {mount_point}", shell=True, sudo=True
+            )
+            mount_result.assert_exit_code(expected_exit_code=32)
+
+    @TestCaseMetadata(
+        description="""
+        This test case will do run commands 2-5, the commands are expected fail or not
+         based on device supports or not.
+        1. Use `nvme id-ctrl device` command list the capabilities of the device.
+        2. `nvme format namespace` - format a namespace.
+        3. `nvme create-ns namespace` - create a namespace.
+        4. `nvme delete-ns -n 1 namespace` - delete a namespace.
+        5. `nvme detach-ns -n 1 namespace` - detach a namespace.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_features=[Nvme],
+        ),
+    )
+    def nvme_manage_nm_validation(self, node: Node) -> None:
+        nvme = node.features[Nvme]
+        nvme_namespaces = nvme.get_namespaces()
+        nvme_devices = nvme.get_devices()
+        nvme_cli = node.tools[Nvmecli]
+        nm_format_exit_code = 0
+        nm_management_exit_code = 0
+        # 1. Use `nvme id-ctrl device` command list the capabilities of the device.
+        if not nvme_cli.support_nm_format(nvme_devices[0]):
+            nm_format_exit_code = 1
+        if not nvme_cli.support_nm_manage_attach(nvme_devices[0]):
+            # NVMe Status:INVALID_OPCODE(1)
+            nm_management_exit_code = 1
+        for namespace in nvme_namespaces:
+            # 2. `nvme format namespace` - format a namespace.
+            format_namespace = nvme_cli.format_namespace(namespace)
+            format_namespace.assert_exit_code(nm_format_exit_code)
+            # 3. `nvme create-ns namespace` - create a namespace.
+            create_namespace = nvme_cli.create_namespace(namespace)
+            create_namespace.assert_exit_code(nm_management_exit_code)
+            # 4. `nvme delete-ns -n 1 namespace` - delete a namespace.
+            delete_namespace = nvme_cli.delete_namespace(namespace, 1)
+            delete_namespace.assert_exit_code(nm_management_exit_code)
+            # 5. `nvme detach-ns -n 1 namespace` - detach a namespace.
+            detach_namespace = nvme_cli.detach_namespace(namespace, 1)
+            detach_namespace.assert_exit_code(nm_management_exit_code)
```

Comment:
Update the `assert_exit_code`, to make it accept a customized message string, and join it with other information.